packages/playground/website/src/components/error-report-modal/index.tsx

```diff
@@ -0,0 +1,101 @@
+import { useEffect, useState } from 'react';
+import Modal from '../modal';
+import { addFatalErrorListener, logger } from '@php-wasm/logger';
+import { Button, TextareaControl, TextControl } from '@wordpress/components';
+
+import css from './style.module.css';
+
+export function ErrorReportModal() {
+	const [hasError, setHasError] = useState(false);
+	const [text, setText] = useState('');
+	const [logs, setLogs] = useState('');
+	const [url, setUrl] = useState('');
+	const [submitted, setSubmitted] = useState(false);
+
+	useEffect(() => {
+		addFatalErrorListener(logger, (e) => {
+			setHasError(true);
+			setText('');
+			setLogs((e as CustomEvent).detail.logs.join(''));
+			setUrl(window.location.href);
+		});
+	}, []);
+
+	function onClose() {
+		setHasError(false);
+		setText('');
+		setLogs('');
+		setUrl('');
+		setSubmitted(false);
+	}
+
+	function onSubmit() {
+		const data = ['What happened?', text, 'Logs', logs, 'Url', url].join(
+			'\n\n'
+		);
+		console.log(data);
+		// TODO: send data to the server
+		setSubmitted(true);
+	}
+
+	return (
+		<Modal isOpen={hasError} onRequestClose={onClose}>
+			{submitted && (
+				<header className={css.errorReportModalHeader}>
+					<h2>Thank you for reporting the error</h2>
+					<p>
+						We will look into the issue and and open an{' '}
+						<a href="https://github.com/WordPress/wordpress-playground/issues/">
+							issue on GitHub if needed
+						</a>
+						.
+					</p>
+				</header>
+			)}
+			{!submitted && (
+				<>
+					<header className={css.errorReportModalHeader}>
+						<h2>Report error</h2>
+						<p>
+							Playground crashed because of an error. We would
+							appreciate it if you would report this error so that
+							we can fix it.
+						</p>
+						<p>
+							It is possible that the error was caused by PHP code
```

Comment:
Oh, would this modal be displayed when the following code is ran? `<?php throw new Exception();`

Reply:
No, I'm not sure why I thought this text would be useful. I'm removing it now.

If you want to test it [this url](http://localhost:5400/website-server/#%7B%20%22landingPage%22:%20%22/wp-admin/%22,%20%22phpExtensionBundles%22:%20%5B%20%22kitchen-sink%22%20%5D,%20%22features%22:%20%7B%20%22networking%22:%20true%20%7D,%20%22steps%22:%20%5B%20%7B%20%22step%22:%20%22login%22%20%7D,%20%7B%20%22step%22:%20%22writeFile%22,%20%22path%22:%20%22/wordpress/wp-content/mu-plugins/rewrite.php%22,%20%22data%22:%20%22%3C?php%20add_action(%20'init',%20function()%20%7B%20throw%20new%20Exception();%20%7D%20);%22%20%7D%20%5D%20%7D) will return the WordPress _There has been a critical error on this website._ screen.